### ISSUE TITLE
Feature: Test times out after set delay

### ISSUE THREAD

**Issue by JordanMartinez** [TestFX/TestFX#397]
Is it possible for the test to throw an unhandled `TimeoutException`, causing the test to fail, after a set delay? In other words:
````java
public class Test extends ApplicationTest {

    @Override
    public void start(Stage primaryStage) throws Exception {
        /* set up test code */
        primaryStage.show();

        // from ReactFX
        // runs the given runnable after the set delay
        FxTimer.runLater(
            Duration.ofSeconds(10), 
            // currently can't do: need to handle exception
            () -> throw new TimeoutException("Test timed out after 10 seconds")
        );
    }

    @Test
    public void doTest() {
        push(MouseButton.PRIMARY); // does something in test

        // if this test hangs for some reason for longer than 10 seconds, 
        //  it should throw the above TimeoutException

        assertTrue(someCondition);
    }

**Comment by brcolow** [comment]
Is the [junit timeout rule](https://github.com/junit-team/junit4/wiki/timeout-for-tests) sufficient for your use case?

**Comment by JordanMartinez** [comment]
Yes! I didn't know about that. Thanks!

**Comment by JordanMartinez** [comment]
To finish my PR in TomasMikula/RichTextFX#504 and in my process of trying to narrow down when the Linux builds in #396 fail and succeed depending on the Travis config file (when the OSX builds are excluded), adding this rule, not the changes I've made to the config file, seems to screw up the build. 

In my tests, I use a base class that extends `ApplicationTest` in the `junit4` module/package that adds a 5 second timeout rule, so that all other subclasses do not need to do this themselves. I wonder if there's a concurrency bug somewhere in test complexity I've built (TestFX + NestedRunner + JUnit4's Timeout)....

So, I'll need to rerun my attempts to figure out what causes the Linux builds to succeed/fail when OSX is excluded, so that I can add the OSX build to the matrix and only it, rather than the Linux ones, fail. Then, I can work on isolating the issue with the OSX build.
